pkg/agent/controller/noderoute/node_route_controller.go

```diff
@@ -548,21 +549,32 @@ func (c *Controller) addNodeRoute(nodeName string, node *corev1.Node) error {
 		return err
 	}
 	peerWireGuardPublicKey := node.Annotations[types.NodeWireGuardPublicAnnotationKey]
+	podCIDRStrs := getPodCIDRsOnNode(node)
+	if len(podCIDRStrs) == 0 {
+		// If no valid PodCIDR is configured in Node.Spec, return immediately.
+		return nil
+	}
```

Comment:
Should the previous routes be deleted if there are? otherwise the following situation may cause an issue:
1. Node A (uid 1) had a PodCIDR 10.0.0.0/24;
2. Node A (uid 1) was deleted, PodCIDR 10.0.0.0/24 was released;
3. Node B was created, PodCIDR 10.0.0.0/24 was allocated to Node B;
4. Node A (uid 2) was created, no PodCIDR was allocated to it.

Node C observed two events: Node B was updated with PodCIDR 10.0.0.0/24, Node A was updated with empty PodCIDR.

When reconciling Node B, the controller won't create new route to 10.0.0.0/24 because of the following `nodesHaveSamePodCIDR` check.
When reconciling Node B, the controller won't delete old route to 10.0.0.0/24 because it returns immediately.

Reply:
Yes, you're right that's an issue.
I think I was assuming that a Node wouldn't stay without PodCIDR, and that it would be resolved, but I guess this situation can last if the cluster has run out of Node CIDRs.